elastic/kibana
# [Cases] Remove case info from alerts when deleting an alert attachment

## Summary

Users can remove alerts from a case by deleting the whole alert attachment. This PR removes the case id from the alerts when deleting an attachment of type alert. It does not remove the case info from all alerts attached to a case when deleting a case. It also fixes a bug where the success toaster will not show when deleting an attachment.

Related: https://github.com/elastic/kibana/issues/146864, https://github.com/elastic/kibana/issues/140800

## Testing

1. Create a case and attach some alerts to the case.
2. Verify that the alerts table (in security or in o11y) shows the case the alert is attached to. You can enable the cases column by pressing "Fields", searching for "Cases", and then selecting the field.
3. Go to the case and find the alerts' user activity.
4. Press the `...` and press "Remove alerts(s)"
5. Go back to the alert table and verify that the case is not shown in the Cases column for each alert.

Please check that when you remove alert(s), attachments (ml, etc), and comments you get a success toaster with the correct text.

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
